### ISSUE TITLE
Docker stops/crashes when running model.train() 

### ISSUE THREAD

**Issue by superqd** [ThilinaRajapakse/simpletransformers#1324]
I am running a simple training test, which runs fine on my Mac, but when I run it inside a docker container, the container just stops/crashes once the training starts (the progress bar and eval loss show, then it just stops).  It just goes back to the command line with no messages or errors whatsoever.  If I run the same code outside of docker, it works fine.

I've setting use_multiprocessing=False, and the dataloader_num_workers=1 in case there was some odd threading issue, but either way causes the same result.  I tried also with the silent=False, but it just dies that way too.

For the docker container, I've tried using a base image of Ubuntu and the one below from MS, but get the same result either way.

Have people had issues with docker before? 

**To Reproduce**
docker: 20.10.10 (I've attached the docker file)
Mac OS: 12.0.1
docker run cotainer_id
latest version of simple transformers (0.63.3)

Here is the output when running the container:
```
Downloading: 100%|██████████| 0.98k/0.98k [00:00<00:00, 670kB/s]
Downloading: 100%|██████████| 420M/420M [00:08<00:00, 50.5MB/s] 
Some weights of the model checkpoint at nlpaueb/bert-base-uncased-contracts were not used when initializing BertForSequenceClassification: ['cls.predictions.transform.dense.weight', 'cls.seq_relationship.weight', 'cls.seq_relationship.bias', 'cls.predictions.bias', 'cls.predictions.decoder.bias', 'cls.predictions.transform.LayerNorm.weight', 'cls.predictions.transform.dense.bias', 'cls.predictions.decoder.weight', 'cls.predictions.transform.LayerNorm.bias']
- This IS expected if you are initializing BertForSequenceClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing BertForSequenceClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
Some weights of BertForSequenceClassification were not initialized from the model checkpoint at nlpaueb/bert-base-uncased-contracts and are newly initialized: ['classifier.weight', 'classifier.bias']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
Downloading: 100%|██████████| 226k/226k [00:00<00:00, 1.24MB/s]
  0%|          | 1/459 [00:00<00:44, 10.39it/s]
Epoch 1 of 5:   0%|          | 0/5 [00:00<?, ?it/s]
MacBook-Pro-2 src % 
```

Here is what the code looks like that is getting executed:
```
model_args = ClassificationArgs(        
        overwrite_output_dir=True,
        best_model_dir="outputs/best_model",
        logging_steps=1,
        save_eval_checkpoints=False,
        early_stopping_consider_epochs=True,
        early_stopping_patience=2,
        use_early_stopping=True,
        num_train_epochs=NUM_TRAINING_EPOCHS,
        learning_rate=LEARNING_RATE,
        #use_multiprocessing=False,
        #dataloader_num_workers=1,
        #silent=False
    )

    # Create a ClassificationModel
    model = ClassificationModel(
        'bert',
        'nlpaueb/bert-base-uncased-contracts',
        use_cuda=False,
        num_labels=2, 
        args=model_args
    )

    # start the training
    model.train_model(train)
```

Here is the docker file:

```
FROM mcr.microsoft.com/vscode/devcontainers/python:0-3.8

RUN curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | bash
RUN python3 -m pip install --upgrade pip

# install libraries (one line at a time mainly for easier to view progress)
RUN pip3 install --no-cache-dir awscli
RUN pip3 install --no-cache-dir Cython
RUN pip3 install --no-cache-dir jsonlines
RUN pip3 install --no-cache-dir pymupdf
RUN pip3 install --no-cache-dir psycopg2-binary
RUN pip3 install --no-cache-dir asyncio
RUN pip3 install --no-cache-dir asyncpg
RUN pip3 install --no-cache-dir pdfplumber
RUN pip3 install --no-cache-dir numpy
RUN pip3 install --no-cache-dir scipy
RUN pip3 install --no-cache-dir scikit-learn
RUN pip3 install --no-cache-dir nltk
RUN pip3 install --no-cache-dir sklearn
RUN pip3 install --no-cache-dir xgboost
RUN pip3 install --no-cache-dir keras
RUN pip3 install --no-cache-dir wandb
RUN pip3 install --no-cache-dir pandas
RUN pip3 install --no-cache-dir torch
RUN pip3 install --no-cache-dir simpletransformers
RUN pip3 install --no-cache-dir sentence-transformers

# get the NLTK stuff downloaded
RUN python3 -m nltk.downloader -d /usr/nltk_data punkt

# preload the Sentence Transfomer base model(s)
RUN python3 -c "from sentence_transformers import SentenceTransformer; embedder = SentenceTransformer('all-MiniLM-L6-v2')"

COPY ./src/experiment.py .

CMD ["python3", "-u", "experiment.py"]
```
